pkg/target/kinesis.go

```diff
@@ -35,48 +35,56 @@ const (
 
 // KinesisTargetConfig configures the destination for records consumed
 type KinesisTargetConfig struct {
-	StreamName        string `hcl:"stream_name" env:"TARGET_KINESIS_STREAM_NAME"`
-	Region            string `hcl:"region" env:"TARGET_KINESIS_REGION"`
-	RoleARN           string `hcl:"role_arn,optional" env:"TARGET_KINESIS_ROLE_ARN"`
-	CustomAWSEndpoint string `hcl:"custom_aws_endpoint,optional" env:"SOURCE_CUSTOM_AWS_ENDPOINT"`
+	StreamName         string `hcl:"stream_name" env:"TARGET_KINESIS_STREAM_NAME"`
+	Region             string `hcl:"region" env:"TARGET_KINESIS_REGION"`
+	RoleARN            string `hcl:"role_arn,optional" env:"TARGET_KINESIS_ROLE_ARN"`
+	RequestMaxMessages int    `hcl:"request_max_messages,optional" env:"TARGET_KINESIS_REQUEST_MAX_MESSAGES"`
+	CustomAWSEndpoint  string `hcl:"custom_aws_endpoint,optional" env:"SOURCE_CUSTOM_AWS_ENDPOINT"`
 }
 
 // KinesisTarget holds a new client for writing messages to kinesis
 type KinesisTarget struct {
-	client     kinesisiface.KinesisAPI
-	streamName string
-	region     string
-	accountID  string
+	client             kinesisiface.KinesisAPI
+	streamName         string
+	region             string
+	accountID          string
+	requestMaxMessages int
 
 	log *log.Entry
 }
 
 // newKinesisTarget creates a new client for writing messages to kinesis
-func newKinesisTarget(region string, streamName string, roleARN string, customAWSEndpoint string) (*KinesisTarget, error) {
+func newKinesisTarget(region string, streamName string, roleARN string, customAWSEndpoint string, requestMaxMessages int) (*KinesisTarget, error) {
 	awsSession, awsConfig, awsAccountID, err := common.GetAWSSession(region, roleARN, customAWSEndpoint)
 	if err != nil {
 		return nil, err
 	}
 	kinesisClient := kinesis.New(awsSession, awsConfig)
 
-	return newKinesisTargetWithInterfaces(kinesisClient, *awsAccountID, region, streamName)
+	// Restrict chunk sizes to the maximum for a PutRecords request, if configured higher.
+	if requestMaxMessages > kinesisPutRecordsMaxChunkSize {
+		requestMaxMessages = kinesisPutRecordsMaxChunkSize
+	}
```

Comment:
Fair point... So, I think on reflection the decision I made here wasn't the best one - I think it's actually better to let the user hit an error because the app did exactly what they configured it to do.

Better still, however, is for it to fail on startup with an error that indicates what's wrong. So that's what I'll do instead I think.